### PR TITLE
[WIP] Add base URL for API version

### PR DESCRIPTION
**Pull request by johngriebel**
feature/issue #161 - Add Base URL for API Version.

### CODE REVIEW

**Comment by johngriebel** [comment]
This pull request implements versioning for the API. As outlined in issue #161, URLs will now be of the form: `https://staging.3blades.io/v1.0/auth/jwt-token-auth/`.

Patch versions are accepted as well:
`https://staging.3blades.io/v1.2.3/auth/jwt-token-auth/`, 
but only the major version is required: 
`https://staging.3blades.io/v1/auth/jwt-token-auth/`.

File: appdj/settings/base.py [hunk 1]
```diff
@@ -231,8 +231,11 @@
         'rest_framework.permissions.IsAuthenticated',
     ),
     'DEFAULT_PAGINATION_CLASS': 'base.pagination.LimitOffsetPagination',
+    'DEFAULT_VERSIONING_CLASS': 'rest_framework.versioning.URLPathVersioning'
 }
 
+DEFAULT_VERSION = "v1"
```
Comment (johngriebel) [inline appdj/settings/base.py]: This will become an environment variable that will be added to the documentation as well.

File: README.md [hunk 2]
```diff
@@ -130,6 +130,7 @@ Modify environment variables located in `env` file with your local settings. You
     SLACK_KEY=
     SLACK_SECRET=
     STRIPE_SECRET_KEY=
+    TBS_DEFAULT_VERSION=v1
```
Comment (johngriebel) [inline README.md]: Maybe this default to blank?

File: appdj/settings/base.py [hunk 3]
```diff
@@ -231,8 +231,11 @@
         'rest_framework.permissions.IsAuthenticated',
     ),
     'DEFAULT_PAGINATION_CLASS': 'base.pagination.LimitOffsetPagination',
+    'DEFAULT_VERSIONING_CLASS': 'rest_framework.versioning.URLPathVersioning'
 }
 
+DEFAULT_VERSION = os.environ.get('TBS_DEFAULT_VERSION', "v1")
+
```
Comment (johngriebel) [inline appdj/settings/base.py]: Maybe `DEFAULT_VERSION` shouldn't have a fallback value? See my other comment.

**Comment by jkp85** [comment]
@johngriebel I wouldn't go to patch change (v1.2.3). Semver definition of patch: `PATCH version when you make backwards-compatible bug fixes`. So it's fixing things that are not working properly. In my opinion we should go with MAJOR.MINOR since minor is `when you add functionality in a backwards-compatible manner`. That being said it really depends on maintainability. If patch is not much a fuss, then I would leave it as is.

**Comment by johngriebel** [comment]
@jkp85 patch version is no fuss at all; it's just a matter of having that part of the regex in the URL meaning there will be no maintenance overhead for including it. Patch version is also entirely optional for clients. I added it mostly so that it will work if a client does include the patch version for whatever reason.

File: appdj/urls/base.py [hunk 4]
```diff
@@ -0,0 +1,6 @@
+from django.conf.urls import url, include
+from django.conf import settings
+urlpatterns = [
+    url(r'^(?P<version>{major_version}\.?[0-9]*\.?[0-9]*)/'.format(major_version=settings.DEFAULT_VERSION),
```
Comment (raygesualdo) [inline appdj/urls/base.py]: `r'^(?P<version>{major_version}((\.[0-9]*){1,2})?'`

File: README.md (hunk 2 above)
Comment (jgwerner) [inline README.md]: We should default to blankin env file. Loader could default to current version if key value is empty.

**Pull request merged**